great_expectations/checkpoint/checkpoint.py

```diff
@@ -425,7 +417,7 @@ def _run_validation(  # noqa: PLR0913
 
             action_list_validation_operator: ActionListValidationOperator = (
                 ActionListValidationOperator(
-                    data_context=self.data_context,
+                    data_context=context,
```

Comment:
Biggest issue is this one here - we need a full context as a dep for validation operators. We'll need a refactoring effort to rip that out before we can update this here.